src/django_rich/runner.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import sys
+import unittest
+from unittest.result import STDERR_LINE, STDOUT_LINE
+
+from django.test import testcases
+from django.test.runner import DebugSQLTextTestResult, DiscoverRunner, PDBDebugResult
+from rich.color import Color
+from rich.console import Console
+from rich.rule import Rule
+from rich.style import Style
+from rich.traceback import Traceback
+
+
+class RichMixin(unittest.TextTestResult):
+    # Traceback width defaults to 100
+    console = Console(width=100, stderr=True)
+    django_green = Style(color=Color.from_rgb(32, 170, 118))
+
+    def addError(self, test, err):
+        super(unittest.TextTestResult, self).addError(test, err)
+        if self.showAll:
+            self.console.print("ERROR", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("E", style=Style(color="red"), end="")
+
+    def addSuccess(self, test):
+        super(unittest.TextTestResult, self).addSuccess(test)
+        if self.showAll:
+            self.console.print("ok", style=self.django_green)
+        elif self.dots:
+            self.console.print(".", style=self.django_green, end="")
+
+    def addFailure(self, test, err):
+        super(unittest.TextTestResult, self).addFailure(test, err)
+        if self.showAll:
+            self.console.print("FAIL", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("F", style=Style(color="red"), end="")
+
+    def addSkip(self, test, reason):
+        super(unittest.TextTestResult, self).addSkip(test, reason)
+        if self.showAll:
+            self.console.print(f"skipped {reason!r}", style=Style(color="yellow"))
```

Comment:
The `reason` for some reason I can't fathom our prints out as green when tests are run in verbose `-v2` mode. May need to ping for help.

Reply:
Perhaps because the `Console` is performing highlighting? https://rich.readthedocs.io/en/stable/highlighting.html

I think this can probably be disabled? Or just left if it's not so problematic.

Reply:
![image](https://user-images.githubusercontent.com/39445562/154565552-281124f9-ebbf-4dbf-aa19-2f2333b632eb.png)

I don't think it's that problematic. Could also ask why unexpected success is lower case etc. (I've copied from unittest)

Reply:
Yeah - in fact it's probably a good thing it's different since it's the reason from the test, as opposed to the testing framework.